firmware/quadruna/CRIT/test/test_ledStatus.cpp

```diff
@@ -0,0 +1,159 @@
+    LetTimePass(10);
+    ASSERT_EQ(1, fake_io_rgbLed_enable_callCountForArgs(&vc_status_led, false, false, true));
+}
+
+TEST_F(LedStatusTest, vc_board_status_led_control_with_no_error)
+{
+    // Don't set any error and check that the DCM LED turns green
+    LetTimePass(10);
+    ASSERT_EQ(1, fake_io_rgbLed_enable_callCountForArgs(&vc_status_led, false, true, false));
+}
+
+TEST_F(LedStatusTest, vc_board_status_led_control_with_multiple_errors)
+{
+    // If the error table contains critical and non-critical errors
+    // simultaneously, the critical error should take precedence and turn the
+    // DCM LED red rather than blue
+    app_canRx_VC_Fault_MissingBMSHeartbeat_update(true);
+    app_canRx_VC_Warning_StackWaterMarkHighTask1Hz_update(true);
+    LetTimePass(10);
+    ASSERT_EQ(1, fake_io_rgbLed_enable_callCountForArgs(&vc_status_led, true, false, false));
+    ASSERT_EQ(0, fake_io_rgbLed_enable_callCountForArgs(&vc_status_led, false, false, true));
+}
+
+TEST_F(LedStatusTest, fsm_board_status_led_control_with_critical_error)
+{
+    // Set any critical error and check that the FSM LED turns red
+    app_canRx_FSM_Fault_MissingBMSHeartbeat_update(true);
+    LetTimePass(10);
+    ASSERT_EQ(1, fake_io_rgbLed_enable_callCountForArgs(&fsm_status_led, true, false, false));
+}
+
+TEST_F(LedStatusTest, fsm_board_status_led_control_with_warning)
+{
+    // Set any warning and check that the FSM LED turns blue
+    app_canRx_FSM_Warning_StackWaterMarkHighTask1Hz_update(true);
+    LetTimePass(10);
+    ASSERT_EQ(1, fake_io_rgbLed_enable_callCountForArgs(&fsm_status_led, false, false, true));
+}
+
+TEST_F(LedStatusTest, fsm_board_status_led_control_with_no_error)
+{
+    // Don't set any error and check that the FSM LED turns green
+    LetTimePass(10);
+    ASSERT_EQ(1, fake_io_rgbLed_enable_callCountForArgs(&fsm_status_led, false, true, false));
+}
+
+TEST_F(LedStatusTest, fsm_board_status_led_control_with_multiple_errors)
+{
+    // If the error table contains critical and non-critical errors
+    // simultaneously, the critical error should take precedence and turn the
+    // FSM LED red rather than blue
+    app_canRx_FSM_Warning_StackWaterMarkHighTask1Hz_update(true);
+    app_canRx_FSM_Fault_MissingBMSHeartbeat_update(true);
+
+    LetTimePass(10);
+    ASSERT_EQ(1, fake_io_rgbLed_enable_callCountForArgs(&fsm_status_led, true, false, false));
+    ASSERT_EQ(0, fake_io_rgbLed_enable_callCountForArgs(&fsm_status_led, false, false, true));
+}
+
+TEST_F(LedStatusTest, bms_board_status_led_control_with_fault)
+{
+    // Set OK statuses such that the red led is not set without fault
+    app_canRx_BMS_ImdOk_update(true);
+    app_canRx_BMS_BspdOk_update(true);
+    app_canRx_BMS_BmsOk_update(true);
+
+    // Set any critical error and check that the BMS LED turns red
+    app_canRx_BMS_Fault_CellOvertemp_update(true);
+    LetTimePass(10);
+
+    ASSERT_EQ(1, fake_io_rgbLed_enable_callCountForArgs(&bms_status_led, true, false, false));
+}
```

Comment:
test all the LEDs too, very good for regression and very important for safety.
SHDN especially.

possibly a very good idea to write a generic test, and pass in function pointers for leds and canerrors as arguments.

very strange that only VC/FSM/2 of CRIT is tested only

make sure that test ordering is also organized and logical.